client/my-sites/checkout/composite-checkout/components/checkout-help-link.tsx

```diff
@@ -193,36 +198,50 @@ export default function CheckoutHelpLink() {
 
 	const hasDirectSupport = supportVariation !== SUPPORT_FORUM;
 
+	const getZendeskChatWidget = () => {
+		if ( isAkismetCheckout() ) {
+			return <ZendeskJetpackChat keyType="akismet" />;
+		}
+
+		if ( isJetpackCheckout() ) {
+			return <ZendeskJetpackChat keyType="jpCheckout" />;
+		}
+
+		if ( isPresalesZendeskChatEligible ) {
+			return (
+				<AsyncLoad
+					require="calypso/components/presales-zendesk-chat"
+					chatKey={ zendeskPresalesChatKey }
+				/>
+			);
+		}
+	};
+
 	// If pre-sales chat isn't available, use the inline help button instead.
 	return (
 		<CheckoutHelpLinkWrapper>
 			<QuerySupportTypes />
 			{ ! isPresalesZendeskChatEligible && ! supportVariationDetermined && <LoadingButton /> }
-			{ isPresalesZendeskChatEligible ? (
-				<AsyncLoad
-					require="calypso/components/presales-zendesk-chat"
-					chatKey={ zendeskPresalesChatKey }
-				/>
-			) : (
-				supportVariationDetermined && (
-					<CheckoutSummaryHelpButton onClick={ handleHelpButtonClicked }>
-						{ hasDirectSupport
-							? translate( 'Questions? {{underline}}Ask a Happiness Engineer{{/underline}}', {
-									components: {
-										underline: <span />,
-									},
-							  } )
-							: translate(
-									'Questions? {{underline}}Read more about plans and purchases{{/underline}}',
-									{
+			{ isPresalesZendeskChatEligible || isJpPresalesStaffed
+				? getZendeskChatWidget()
+				: supportVariationDetermined && (
```

Comment:
This logic is somewhat difficult to parse.

Is there a way we can make this clearer/ easier to follow?

I notice also that using `isPresalesZendeskChatEligible || isJpPresalesStaffed` opens up some questions like "What happens if `isJpPresalesStaffed` is true and `isPresalesZendeskChatEligible` is false?" I can derive the answer from looking at `getZendeskChatWidget ` but since the logic is spread out it seems a bit more fragile.

Reply:
If `isJpPresalesStaffed `is true, then `getZendeskChatWidget` will check if the page is Akismet checkout or Jetpack checkout, and display the correct chat widget there. If it is neither of those pages, nothing will happen

Reading through this, I am actually seeing an issue if 'isJpPresalesStaffed` is false and `isPresalesZendeskChatEligible` is true, where if on the akismet or jetpack checkout, it displaying the wordpress chat widget. 

I do think the logic needs some work, apologies for not catching that earlier. I'll also see if I can get it a bit more clear to read

Reply:
I tried to make the logic a bit more readable as well as add a few comments. I believe I also fixed up my logic hole mentioned above